### PR TITLE
Update repo to use plugin as a library

### PR DESCRIPTION
**Pull request by colinhoglund**
This refactors the plugin package to make it more friendly to use as a library.

Changes:
- Embed the default policy in the plugin package so a library user is not required to load policy configuration
- Update test to reflect new package API
- Move log format and output configuration to `cli` package
- Removing drone pipeline and no longer publishing a container image to quay
- Update to Go 1.21
- Update dependencies
- Remove usage of `ioutil`
- Update instructions in readme

An example of usage can be seen in this branch of the drone-extension-router: https://github.com/kanopy-platform/drone-extension-router/compare/main...add_opa_validation_plugin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,15 +43,16 @@ Create a shared secret:
...
-docker run -d \
+docker build -t drone-validation .
+
+docker run \
   --publish=3000:3000 \
   --env=DRONE_DEBUG=true \
-  --env=DRONE_SECRET=<your_shared_secret> \
-  --restart=always \
-  --name=<container_name> quay.io/mongodb-labs/drone-validation:latest 
+  --env=DRONE_VALIDATE_PLUGIN_SECRET=$(openssl rand -hex 16) \
```
Comment (lviana) [inline README.md]: The same secret needs to be configured in drone-server. If we suggest creating a new one dynamically every time the plugin starts, users will have to discover the runtime password by themselves.

Comment (colinhoglund) [reply README.md]: This makes this a working example to run locally. This would not be a valid way to run in any real world deployment.

Comment (colinhoglund) [reply README.md]: changed the wording of the readme

**Review by dskatz — DISMISSED** [review]
LGTM.

minor nit:  should the `plugin` directory be moved into `pkg/plugin`?

**Review by dskatz — APPROVED** [review]

**Pull request merged**